content/courses/onchain-development/local-setup.md

```diff
@@ -2,116 +2,219 @@
+**The `anchor test` command should complete without errors or warnings**. If you
+encounter issues, we’ll address them below:
+
+#### Error: `package 'solana-program v1.18.12' cannot be built because it requires rustc 1.75.0 or newer`
 
 Run `cargo add solana-program@"=1.18.x"`, where `x` matches your version of
 `solana-cli`. Then re-run `anchor test`.
 
-##### `Error: Unable to read keypair file`
+#### Error: `Unable to read keypair file`
 
 Add a keypair to `.config/solana/id.json`. You can either copy a keypair from an
 `.env` file (just the array of numbers) into a file or use the command
 `solana-keygen new --no-bip39-passphrase` to create a new keypair file. Then
 re-run `anchor test`.
 
-##### `unused variable: 'ctx'` warning
+#### Error: `Function _ZN112_$LT$solana_program..instruction..InstructionError$u20$as$u20$solana_frozen_abi..abi_example..AbiEnumVisitor$GT$13visit_for_abi17ha3a75fe06f2a2af4E Stack offset of 4584 exceeded max offset of 4096 by 488 bytes, please minimize large stack variables`
+
+If you encounter the error above, follow these steps to resolve it:
+
+1. **Update Dependencies**: Run the following command to update the dependencies
+   in your `Cargo.lock` file to the latest versions. If the `Cargo.lock` file
+   does not exist, this command will create it with the latest available
+   versions.
+
+   ```bash
+   cargo update
+   ```
+
+2. **Upgrade solana-program**: Please upgrade to `solana-program` version
+   `1.18.2` or above, which includes the necessary fix. You can update to the
+   latest stable version using one of the following methods:
+
+   - **Using Cargo Command**: Run the following command to update
+     solana-program:
 
-This simply means the `initialize` instruction handler isn't doing anything yet.
-You can open `programs/favorites/src/lib.rs` and change `ctx` to `_ctx` or just
-go onto the next step.
+     ```bash
+     cargo add solana-program
+     ```
 
-##### `No license field in package.json` warning
+   - **Manually Update Cargo.toml**: Alternatively, add the following line to
+     your Cargo.toml file:
 
-Open package.json, add `"license": "MIT"` or `"license": "UNLICENSED"` depending
-on preferences
+     ```bash
+     solana-program = "2.0.8"
+     ```
+
+   You can find the latest stable version on the
+   [solana-program crate page](https://crates.io/crates/solana-program).
+
+3. **Re-run Tests**: After upgrading, re-run your tests with the following
+   command:
+
+   ```bash
+   anchor test
+   ```
+
+#### Warning: `unused variable: 'ctx'`
+
+This warning indicates that the `initialize` instruction handler isn’t doing
+anything yet. You can either open `programs/favorites/src/lib.rs` and change
+`ctx` to `_ctx` or proceed to the next step.
+
+#### Warning: `No license field in package.json`
+
+Open `package.json`, add `"license": "MIT"` or `"license": "UNLICENSED"`
+depending on preferences.
```

Comment:
You can delete this too, I fixed it in `anchor init` in Anchor 0.30.1 (Anchor now uses your default license)